src/accumulator.jl

```diff
@@ -24,6 +24,14 @@ function counter{T}(seq::AbstractArray{T})
     return ct
 end
 
+function counter{T<:Base.Generator}(gen::T)
+    ct = counter(Base._default_eltype(T))
```

Comment:
I wonder if there is a better way to do this?  My only concern is that this is meant to be an internal function and could (in theory) go away...

Anyway, this is useful functionality.  Sorry to let it languish!

Reply:
I wondered that too, but it's how [`Base.Set`](https://github.com/JuliaLang/julia/blob/b46c74cc9c61769aade33e407c202bf88aaf39d1/base/set.jl#L20) does it, and similarly for `Dict`. It's defined as `_default_eltype(itrt::ANY) = Core.Inference.return_type(first, Tuple{itrt})`, so I don't think we can do it correctly using exported functionality.